### ISSUE TITLE
[bug] Rendering to different HTML than input

### ISSUE THREAD

**Issue by gabyx** [executablebooks/mdformat#292]
### Describe the problem

Error: Could not format "spec.md".

The formatted Markdown renders to different HTML than the input Markdown. This
is likely a bug in mdformat. Please create an issue report here, including the
input Markdown: https://github.com/executablebooks/mdformat/issues

### Steps to reproduce

Format the attached file. It is the Google C++ Convention (converted by pandoc etc)

[spec.md](https://github.com/executablebooks/mdformat/files/7807461/spec.md)

### The version of Python you're using

3.9

### Your operating system

MacOs

### Versions of your packages

```
❯ pip list          
Package            Version
------------------ ----------
aiofiles           0.6.0
aiohttp            3.7.4
async-timeout      3.0.1
attrs              20.3.0
bikeshed           3.3.4
binaryornot        0.4.4
bottle             0.12.19
certifi            2020.12.5
chardet            3.0.4
charset-normalizer 2.0.7
click              8.0.3
cmake-format       0.6.13
cmakelang          0.6.13
colorama           0.4.4
commentjson        0.9.0
conan              1.43.0
cssselect          1.1.0
distro             1.6.0
fasteners          0.16.3
filelock           3.3.2
fire               0.4.0
flake8             4.0.1
html5lib           1.1
idna               2.10
importlib-metadata 4.8.2
isodate            0.6.0
Jinja2             2.11.3
json-home-client   1.1.1
lark-parser        0.7.8
lxml               4.6.3
markdown-it-py     1.1.0
MarkupSafe         2.0.1
mccabe             0.6.1
mdformat           0.7.10
multidict          5.2.0
node-semver        0.6.1
packaging          21.2
panflute           2.1.3
patch-ng           1.17.4
Pillow             8.1.1
pip                21.3.1
pluginbase         1.0.1
pycodestyle        2.8.0
pydocstyle         6.1.1
pyflakes           2.4.0
Pygments           2.8.1
PyJWT              1.7.1
pylama             8.3.6
pyparsing          2.4.7
python-dateutil    2.8.2
PyYAML             5.4.1
regex              2021.11.10
requests           2.25.1
result             0.6.0
setuptools         57.4.0
six                1.16.0
snowballstemmer    2.2.0
tenacity           7.0.0
termcolor          1.1.0
tomli              1.2.2
tqdm               4.62.3
typing_extensions  4.0.0
uri-template       1.1.0
urllib3            1.26.7
webencodings       0.5.1
wheel              0.37.0
widlparser         1.0.12
yapf               0.31.0
yarl               1.7.2
zipp               3.6.0```

**Comment by hukkin** [comment]
Thanks for the issue!

Here's something smaller that also reproduces the error:

```markdown
A
    <div>
    </div>
```

**Comment by hukkin** [comment]
If you want to format the file before we have proper fix, you can do the following:

Search the file for a pattern where a paragraph is followed by an indented `<div>` tag and there is no empty line in between. Remove indentation from the `<div>` tag and its closing `</div>` tag. This pattern occurs in your document about 5 times, so its possible to do this by hand. After this change the file should format just fine.

For example, change this:
```markdown
    specifying the type of either the enclosing object or the individual names:
    <div class="neutralcode">

    ``` cpp
    auto [iter, success] = my_map.insert({key, value});
    if (!success) {
      iter->second = value;
    }
    ```

    The

    </div>
```
to this
```markdown
    specifying the type of either the enclosing object or the individual names:
<div class="neutralcode">

    ``` cpp
    auto [iter, success] = my_map.insert({key, value});
    if (!success) {
      iter->second = value;
    }
    ```

    The

</div>
```

**Comment by gabyx** [comment]
Thanks.

**Comment by hukkin** [comment]
This should be fixed now in v0.7.12
